enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,310 @@
+
+If `platform.azure.userTags` of install-config has any tag defined same will be added 
+to all the azure resources created by Openshift except, when the tag validation fails 
+due to any of below conditions
+1. A tag name can have a maximum of 128 characters.
+    - Tag name has a limit of 512 characters for all resources except for 
+    storage accounts, which has a limit of 128 characters and hence tag name 
+    length is restricted to 128 characters on every resource required by Openshift.
+2. A tag name cannot contain `<, >, %, &, \, ?, /, #, :, whitespace` characters and 
+   should not start with a number.
+    - DNS zones, Traffic, Front Door resources does not support tag with spaces, 
+    special/unicode characters or starting with number, hence these are added as 
+    constraints on every other Azure resource required by Openshift as well.
+3. A tag value has a limit of 256 characters.
+4. A resource, resource-group or subscription, user can configure a maximum of 5 tags
+   through Openshift. 
+    - Azure supports a maximum of 50 tags except for Automation, Content Delivery Network,
+    DNS resources which can have a maximum of 15 tags, hence restricting the number of 
+    user defined tags to 5 and 10 for Openshift internal use for all resources created 
+    by Openshift.
+
+All in-cluster operators that create Azure resources (Cluster Infrastructure ,Storage ,Node ,NetworkEdge , Internal Registry ,CCO) will apply these tags during resource creation.
+
+The userTags field is intended to be set at install time and is considered immutable. 
+Components that respect this field must only ever add tags that they retrieve from this 
+field to cloud resources, they must never remove tags from the existing underlying cloud 
+resource even if the tags are removed from this field(despite it being immutable).
+
+If the userTags field is changed post-install, there is no guarantee about how an 
+in-cluster operator will respond to the change. Some operators may reconcile the 
+change and change tags on the Azure resource. Some operators may ignore the change. 
+However, if tags are removed from userTags, the tag will not be removed from the 
+Azure resource.
+
+### Workflow Description
+
+- An Openshift administrator requests to add required tags to all Azure resources 
+  created by Openshift by adding it in `.platform.azure.userTags`
+- openshift installer validates the tags defined in `.platform.azure.userTags` and 
+  adds these tags to all resources created during installation and also updates 
+  `.spec.platformSpec.azure.resourceTags` of the `infrastructure.config.openshift.io`
+- In cluster operators refers `.spec.platformSpec.azure.resourceTags` of the 
+  `infrastructure.config.openshift.io` to add tags to resources created later.
+- An Openshift administrator can modify existing tags or add new tags by updating 
+  `.spec.platformSpec.azure.resourceTags` field in the `infrastructure.config.openshift.io`.
```

Comment:
Earlier it is stated "If the userTags field is changed post-install, there is no guarantee about how an in-cluster operator will respond to the change." and the Non-goals imply modification of userTags to be out of scope.  Will there be any restriction on this property to limit unexpected behavior, so that updates require intentional removal of an admission controller?

Reply:
Initially when creating this proposal, had covered update and delete of tags too, but it's out of scope with the current requirement. I have missed updating above section with the same.